api/dp-openapi.yaml

```diff
@@ -733,6 +845,63 @@ components:
       in: header
       name: X-API-KEY
 
+  parameters:
+    projectid:
+      name: projectid
+      in: path
+      description: ID of project
+      required: true
+      schema:
+        type: string
+
+    state:
+      name: state
+      in: query
+      description: Project state
+      required: true
+      schema:
+        type: string
+
+    pagename:
+      name: pagename
+      in: path
+      description: name of page
+      required: true
+      schema:
+        type: string
+
+    pagestate:
+      name: pagestate
+      in: query
+      description: page state
+      required: true
+      schema:
+        type: string
+
+    pageaction:
+      name: pageaction
+      in: query
+      description: action to perform on the page.<br><br>
+        resume If the page is saved-as-done 'unsaves' it, gets data for the given page.<br><br>
+        saveasdone Checks the page for invalid characters and if valid, saves it as done. Else throws an exception.<br><br>
```

Comment:
I would try to hide some of our internal workings from the API client. Consider the following verbs instead (same order as here):
```
checkout - Puts the saved page in a checked-out state for changing
checkin - Saves the page as completed
save - Saves the page and keeps it checked out
revert - Reverts to the original page text
abandon - Returns the page to the round
```

These mirror the current actions but hides the confusing complexity. `checkout` (ie `resume`) pairs with `return` and mirrors how the API got the page to begin with. You can `save` something but until you `checkin` or `abandon` it it's still checked out to you. This way we avoid the confusing `temp` descriptions here -- it's just "checked out" to the user.

Also no need to include "Else throws an exception" as that's a given for any non-success case.

Reply:
I'll do that.

Reply:
done these.

Reply:
I've left resume as it is bedause checkout could be confusing since we already have a checkout function to get a new page. Perhaps we could change the word for that function and use checkout and checkin  for a given page.